azure-iot-device/azure/iot/device/common/auth/sastoken.py

```diff
@@ -94,3 +97,62 @@ def _build_token(self):
     def expiry_time(self):
         """Expiry Time is READ ONLY"""
         return self._expiry_time
+
+
+class NonRenewableSasToken(object):
+    """NonRenewable Shared Access Signature Token used to authenticate a request.
+
+    This token is 'non-renewable', which means that it is invalid once it expires, and there
+    is no way to keep it alive. Instead, a new token must be created.
+
+    Data Attributes:
+    expiry_time (int): Time that token will expire (in UTC, since epoch)
+    resource_uri (str): URI for the resource the Token provides authentication to access
+    """
+
+    def __init__(self, sastoken_string):
+        """
+        :param str sastoken_string: A string representation of a SAS token
+        """
+        self._token = sastoken_string
+        self._token_info = get_sastoken_info_from_string(self._token)
+
+    def __str__(self):
+        return self._token
+
+    @property
+    def expiry_time(self):
+        """Expiry Time is READ ONLY"""
+        return int(self._token_info["se"])
+
+    @property
+    def resource_uri(self):
+        """Resource URI is READ ONLY"""
+        uri = self._token_info["sr"]
+        return urllib.parse.unquote(uri)
+
+
+REQUIRED_SASTOKEN_FIELDS = ["sr", "sig", "se"]
+VALID_SASTOKEN_FIELDS = REQUIRED_SASTOKEN_FIELDS + ["skn"]
+
+
+def get_sastoken_info_from_string(sastoken_string):
+    pieces = sastoken_string.split("SharedAccessSignature ")
+    if len(pieces) != 2:
+        raise SasTokenError("Invalid SasToken string: Not a SasToken ")
+
+    # Get sastoken info as dictionary
+    try:
+        sastoken_info = dict(map(str.strip, sub.split("=", 1)) for sub in pieces[1].split("&"))
+    except Exception as e:
+        raise SasTokenError("Invalid SasToken string: Incorrectly formatted", e)
+
+    # Validate that all required fields are present
+    if not all(key in sastoken_info for key in REQUIRED_SASTOKEN_FIELDS):
+        raise SasTokenError("Invalid SasToken string: Not all required fields present")
+
+    # Validate that no unexpected fields are present
+    if not all(key in VALID_SASTOKEN_FIELDS for key in sastoken_info):
+        raise SasTokenError("Invalid SasToken string: Unexpected fields present")
```

Comment:
Not actually sure this is needed or desirable.  Is there something else in your sas token handling that will break if there were additional fields?